plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -780,6 +780,24 @@ public void testRowLevelDeletesWithTableChangesFunction()
         }
     }
 
+    @Test
+    public void testCreateOrReplaceWithTableChangesFunction()
+    {
```

Comment:
> Note that the exception is not a Trino exception. 

Why wouldn't this be a `TrinoException` ?

Reply:
It is a `IllegalArgumentException`

Reply:
Please add 

```
        if (!SnapshotUtil.isParentAncestorOf(icebergTable, endSnapshotId, startSnapshotId)){
            throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Starting snapshot (exclusive) %s is not a parent ancestor of end snapshot %s".formatted(startSnapshotId, endSnapshotId));
        }
```

after https://github.com/trinodb/trino/blob/875fe1b8a891e3293f0baf48cb4eea7b815b8b3d/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/functions/tablechanges/TableChangesFunction.java#L116-L117